### DISCUSSION TITLE
How Do CALCS get stored when logged into Google Account?

### DISCUSSION THREAD

**Discussion by dbareis** [kazad/instacalc#97] [Q&A]
I am logged in to google in Chrome on Windows 11

I just logged in on Android PWA, and have what I assume is a default sheet.  

After a bit of a panic I restarted W11 chrome and it still has my sheets.  

So am I correct that there is no syncing and therefore also no backup of sheets?  I'd much rather it used Google drive for storage of the sheets.  At a minimum there needs to be a way to backup the data locally ourselves.

I just looked at "**localStorage**" in the w11 browser's profile and it has a poorly named item called "**store**" (poor as someone else might also use that), which seems to contain the JSON, is this the only place it is stored?  If so it would be a trivial process to create a button to save and restore state....

**Comment by kazad** [comment]
Hi dbareis, each browser session is independent. But if you log in with Google (on each device), it will sync calcs between shared sessions. So, if you log into your Windows 11 computer, and also into your Android PWA, it will sync calcs between them (and you'll see the combined list on each device).

Luckily, browsers isolate localStorage by domain, so the `store` element used by `instacalc.com` can't be read or written to by other websites.

I'd like to have a file import/export, so you can store calcs locally on your desktop as well. Browsers are coming up with recent APIs to make that easier (https://developer.mozilla.org/en-US/docs/Web/API/File_System_API/Origin_private_file_system). I have an internal note to make a quick calc export/import UI so you can easily back up calcs as well.

**Comment by dbareis** [comment]
I have created a small bookmarklet to backup/export the data (attached), now that you have confirmed where the the data is stored, I will create an import function.

[Save InstaCalc Storage.zip](https://github.com/user-attachments/files/16123245/Save.InstaCalc.Storage.zip)

And again when I logged into my google account on a different computer, I did not get my existing calcs....

**Comment by dbareis** [comment]
It looks like to read files there needs to be user interaction so it can only be done by your extension.  With Chrome Debug tools I will be able to manually do it for now if required.

**Comment by kazad** [comment]
I've just added the import/export, details here: https://github.com/kazad/instacalc/discussions/98

For the calc sync, if you visit https://instacalc.com/account.html, do you see the same userID in both browsers? Something like:

`user:  xxxxxxx`

**Comment by dbareis** [comment]
I assume you mean "UID" as I don't see "user:".  I will check next time it occurs.
